Source/NSTableView.m

```diff
@@ -2338,28 +2341,35 @@ - (void) setDataSource: (id)anObject
 
   // If we have content binding the data source is used only
   // like a delegate
-  theBinding = [GSKeyValueBinding getBinding: NSContentBinding 
-                                  forObject: self];
+  theBinding = [GSKeyValueBinding getBinding: NSContentBinding
+				  forObject: self];
   if (theBinding == nil)
-    { 
-      if (anObject && [anObject respondsToSelector: sel_a] == NO) 
-        {
-          [NSException 
-            raise: NSInternalInconsistencyException 
-            format: @"Data Source doesn't respond to numberOfRowsInTableView:"];
-        }
-      
-      if (anObject && [anObject respondsToSelector: sel_b] == NO) 
-        {
-          /* This method isn't required.
-             [NSException raise: NSInternalInconsistencyException 
-             format: @"Data Source doesn't respond to "
-             @"tableView:objectValueForTableColumn:row:"];
-          */  
-        }
-    }
-
-  _dataSource_editable = [anObject respondsToSelector: sel_c];
+    {
+      if (anObject && [anObject respondsToSelector: sel_a] == NO)
+	{
+	  [NSException
+	    raise: NSInternalInconsistencyException
+	    format: @"Data Source doesn't respond to numberOfRowsInTableView:"];
+	}
+
+      if (anObject && [anObject respondsToSelector: sel_b] == NO)
+	{
+	  /* This method isn't required.
+	     [NSException raise: NSInternalInconsistencyException
+	     format: @"Data Source doesn't respond to "
+	     @"tableView:objectValueForTableColumn:row:"];
+	  */
+	}
+
+      _dataSource_editable = [anObject respondsToSelector: sel_c];
+    }
+  else
+    {
+      /* Based on testing on macOS, this should default to YES if there is a binding...
```

Comment:
If this is true, you don't have to check for the binding in line 3338. Looking at this variable should be enough.

Reply:
Removed

Reply:
Both instances.